docs/DeveloperGuide.md

```diff
@@ -1,29 +1,265 @@
+5. After each placement, the game would check the status to determine whether continue the game or not.
+6. If the board is full or there are three consecutive marks, the game will announce the winner or say "it's a draw".
+7. Finally, the `ExerciseCommand` would update the player profile with the reward or punishment for health.
+
+## MiniGame - True or False
+
+1. The game can be invoked by the `RestCommand` class when the user inputs the `rest` command.
+2. It makes use of the `ResponseManager` for instructions and display the questions to answer.
+3. The user would be prompted to answer True or False by typing `T` or `F`.
+4. When the user returns his/her answer, the `ResponseManager` will display the correctness of the response.
+5. When the user finishes answering all the questions, `ResponseManager` will output how many questions the user answers correctly.
+6. Finally, the `RestCommand` would update the player profile with the reward or punishment earned according to the game.
+
+## Investment game - Stock
+
+1. The process of buying stock can be invoked by the `StockCommand` class when user inputs `stock`.
+2. It makes use of `ResponseManager` for instructions and output the stock graph and information relating to the stock.
+3. The user would be prompted to input the number of stocks they would like to purchase in integer.
+4. `ResponseManager` will indicate the completion of purchase after user's input.
+5. When the user finishes the purchase, stock they bought will be stored inside `Asset` class.
+6. Input `status` will output the stock the user currently possesses.
+7. The process of selling stock can be invoked by the `SellStockCommand` class when user input `sellstock`.
+8. The system will calculate user's profit gained at the current price of the stock and return the money back to the 
+user based on the stock's current price.
+9. User can keep the stock they purchased inside their asset for as long as they want.
+
+## Company Management
+1. Players are able to invoke different commands to manage their company.
+2. The corresponding commands will be produced from parsing the user input by using `CommandFactory`.
+3. Then the commands produced will be access and executed by `Command` interface in `EconoCraftLogic`.
+4. Different command would adjust the Company's status such as hire employee, fire employee, and raise salary etc.
+5. At the end of the game round, the company would earn asset to the player based on the company's status.
+
+## Random Event
+1. Random event will be triggered at the end of each round.
+2. The event will be generated by `EventGenerator` class.
+3. The event generated will be executed via `RandomEvent` interface in `EconoCraftLogic`.
+4. Player would be prompted to make a decision based on the event.
+5. Based on the decision made and the event type, the player would receive different rewards or punishments.
+6. The event would add uncertainty and excitement to the game.
 
```

Comment:
It would be nice if the target user for the product and/or the value proposition is specified clearly.